internal/error.go

```diff
@@ -222,6 +222,9 @@ type (
 	// already started.
 	ChildWorkflowExecutionAlreadyStartedError struct{}
 
+	// NamespaceNotFoundError is set as the cause when failure is due namespace not found.
+	NamespaceNotFoundError struct{}
```

Comment:
Could consider reusing `go.temporal.io/api/serviceerror.NamespaceNotFound` instead of adding this, but I don't mind this either (and maybe you don't have the namespace to put on the error at error time).

Reply:
I wouldn't mix these too and prefer to follow "the pattern".